lifecycle/object.go

```diff
@@ -141,6 +142,9 @@ func (o *objectLifecycleAdapter) removeFinalizer(name string, obj runtime.Object
 		}
 
 		obj, err = o.objectClient.GetNamespaced(metadata.GetNamespace(), metadata.GetName(), metav1.GetOptions{})
+		if errors.IsNotFound(err) {
```

Comment:
Would a Debug log be valuable or no ?

Reply:
I think it would just add noise but I don't feel strongly about it either way